xbmc/application/Application.cpp

```diff
@@ -2331,11 +2331,27 @@ bool CApplication::PlayFile(CFileItem item,
   if (PLAYLIST::IsPlayList(item))
     return false;
 
-  // Translate/Resolve the url if needed
-  const std::unique_ptr<IDirectory> dir{CDirectoryFactory::Create(item)};
-  if (dir && !dir->Resolve(item))
+  // Translate/Resolve the url if needed - recursively, but only limited times.
+  static constexpr size_t MAX_ITEM_RESOLVE_ATTEMPTS{5};
+
+  std::string lastDynPath{item.GetDynPath()};
+  size_t itemResolveAttempt{0};
+  while (itemResolveAttempt < MAX_ITEM_RESOLVE_ATTEMPTS)
   {
-    return false;
+    itemResolveAttempt++;
+
+    const std::unique_ptr<IDirectory> dir{CDirectoryFactory::Create(item)};
+    if (dir && !dir->Resolve(item))
+    {
+      CLog::LogF(LOGERROR, "Error resolving item. Item '{}‘ is not playable.", item.GetDynPath());
+      return false;
+    }
+
+    const std::string newDynPath{item.GetDynPath()};
+    if (newDynPath == lastDynPath)
+      break; // done
+
+    lastDynPath = newDynPath;
```

Comment:
Depending on how you feel about saving a copy:
```suggestion
    std::string newDynPath{item.GetDynPath()};
    if (newDynPath == lastDynPath)
      break; // done

    lastDynPath = std::move(newDynPath);
```

Reply:
Good idea, will change.